src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/mappers/FlowMapper.java

```diff
@@ -69,14 +73,18 @@
     @Mapping(target = "meterId", ignore = true)
     @Mapping(target = "transitEncapsulationId", ignore = true)
     @Mapping(target = "diverseWith", ignore = true)
+    @Mapping(target = "mirrorPathStatuses", ignore = true)
     public abstract FlowDto map(Flow flow);
 
     /**
      * Convert {@link Flow} to {@link FlowDto} with diverse flow ids.
      */
-    public FlowDto map(Flow flow, Set<String> diverseWith) {
+    public FlowDto map(Flow flow, Set<String> diverseWith, List<FlowMirrorPath> flowMirrorPaths) {
         FlowDto flowDto = map(flow);
         flowDto.setDiverseWith(diverseWith);
+        flowDto.setMirrorPathStatuses(flowMirrorPaths.stream()
```

Comment:
It should be possible to do via mapstruct.

Reply:
Now mapstruct is used